token/src/instruction.rs

```diff
@@ -85,6 +85,17 @@ pub enum TokenInstruction {
     ///   2. `[writable]` Token being burned.
     ///   3. Optional: `[writable]` Source account if key 1 is a delegate account.
     Burn(u64),
+
+    /// Wrap SOL into a token account.  The full lamport balance in the new account will be wrapped
+    ///   0. `[writable]` New wrapped account being created.
```

Comment:
Note: I dropped the signer requirement (`NewAccount` should do the same IMO).   Requiring a signer here means the token account can't be derived account (aka `CreateAccountWithSeed`).

Reply:
You can wrap my account without permission?

Reply:
Yes but only if you leave it in as `State::Unallocated` (ie, you don't create and initialize your account in an atomic transaction)

Reply:
That footgun should probably be well documented

Reply:
Yeah, I'll add a comment here.   The Rust/JS bindings should do the right thing.   Note that Vote and Stake account initialization behave this way too.

Reply:
I added a comment, also renamed to `InitializeWrappedAccount` since it's not actually creating the account but only initializing it

Reply:
+1 for Initialize. This feels a lot more consistent with our native programs.

Reply:
https://github.com/solana-labs/solana-program-library/pull/70